### PR TITLE
autoconf-archive: Update to 2023.02.20

### PR DESCRIPTION
**Pull request by malfisya**
**Summary**

- Full changelog can be found inside source [tarball](http://ftp.wayne.edu/gnu/autoconf-archive/autoconf-archive-2023.02.20.tar.xz)
- Added `homepage` key to `package.yml` (Part of #411)

**Test Plan**

- Rebuild `mate-desktop` against this package locally
- Verified the added homepage loads information about the package

**Checklist**

- [x] Package was built and tested against unstable

### CODE REVIEW

File: packages/a/autoconf-archive/package.yml [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
-version    : 2019.01.06
-release    : 7
+version    : 2023.02.20
+release    : 8
 source     :
-    - https://ftp.gnu.org/gnu/autoconf-archive/autoconf-archive-2019.01.06.tar.xz : 17195c833098da79de5778ee90948f4c5d90ed1a0cf8391b4ab348e2ec511e3f
-license    : GPL-3.0-with-autoconf-exception
+    - http://ftp.wayne.edu/gnu/autoconf-archive/autoconf-archive-2023.02.20.tar.xz : 71d4048479ae28f1f5794619c3d72df9c01df49b1c628ef85fde37596dc31a33
+homepage   : https://www.gnu.org/software/autoconf-archive/
+license    : GPL-3.0-or-later WITH Autoconf-exception-3.0
```
Comment (TraceyC77) [inline packages/a/autoconf-archive/package.yml:7]: According to [SPDX](https://spdx.org/licenses/GPL-3.0-with-autoconf-exception.html) this license identifier is correct

**Review by TraceyC77 — CHANGES_REQUESTED** [review]
There is a changelog in the source tarball. Please use that to update the summary with the changes and letting users know where to find the complete changelog.

File: packages/a/autoconf-archive/package.yml (hunk 1 above)
Comment (malfisya) [reply packages/a/autoconf-archive/package.yml:7]: Yes, but deprecated isn't it?. 

> DEPRECATED: Use license expression including main license, "WITH" operator, and identifier: Autoconf-exception-3.0

The new one should be the correct one. I am happy to change it back.

Comment (silkeh) [reply packages/a/autoconf-archive/package.yml:7]: No, it isn't: It should be `GPL-3.0-with-autoconf-exception`

**Review by TraceyC77 — DISMISSED** [review]
Source for the source should remain gnu.org

File: packages/a/autoconf-archive/package.yml [hunk 2]
```diff
@@ -1,9 +1,10 @@
 name       : autoconf-archive
-version    : 2019.01.06
-release    : 7
+version    : 2023.02.20
+release    : 8
 source     :
-    - https://ftp.gnu.org/gnu/autoconf-archive/autoconf-archive-2019.01.06.tar.xz : 17195c833098da79de5778ee90948f4c5d90ed1a0cf8391b4ab348e2ec511e3f
-license    : GPL-3.0-with-autoconf-exception
+    - http://ftp.wayne.edu/gnu/autoconf-archive/autoconf-archive-2023.02.20.tar.xz : 71d4048479ae28f1f5794619c3d72df9c01df49b1c628ef85fde37596dc31a33
```
[the remark this answers is not in the document]
Comment (malfisya) [reply packages/a/autoconf-archive/package.yml:5]: This is a gnu mirror, automatically chosen for me based on my region. I use the download link from the homepage. Sure, i can change it if necessary.

**Review by silkeh — APPROVED** [review]
LGTM, thanks! Will land this after the ISO release as this is a build dependency of some pretty major components and anything that *might* impact the release is frozen.

**Pull request merged**